### ISSUE TITLE
Best effort multithreading cleanup version of os._exit()

### ISSUE THREAD

**Issue by Marc-Morcos** [python/cpython#95755]
# Feature or enhancement + Pitch

 I am looking for a way to shut down all threads that provides best effort cleanup. I understand that its not possible to guarantee cleanup since one of the threads might hang in the shutdown process. To address this, a timeout argument can be provided. You tell all the threads to cleanup. If they are not done by the timeout, you do what os._exit() does. This would help with error handling where you want to end a program and a clean shutdown is preferred, but not critical.

# Previous discussion

Related to an old thread  #77268

**Comment by cmaloney** [comment]
There's an active PEP (pep-788) in this area: https://discuss.python.org/t/pep-788-protecting-the-c-api-from-interpreter-finalization/104150

**Comment by picnixz** [comment]
@ZeroIntensity I think your PEP shold cover this so I guess we can close this right?

**Comment by ZeroIntensity** [comment]
Yeah, and the shutdown behavior has improved a lot since 2022, so cleaning up threads is more reliable these days.
